### PR TITLE
Varadero Sprite Selects

### PR DESCRIPTION
**Pull request by Agameofscones**
## About The Pull Request

Cherry picks a bunch of content from Varadero, mostly sprites, and adds them to the game. this includes the PKD series of revolvers from the blade runner series as well since that's canon and based. I'm gonna get a contributor to do the equipment presets for it.

Also minor energy gun refactor, and adds the RXM EVA pistol from the original film.

## Why It's Good For The Game

Old bay12 sprites be gone!

## Changelog

:cl: Triiodine, Monkeysfist101
refactor: Rewrote some energy gun code so we can actually have energy guns now.
fix: Body bag boxes no longer fit into storage containers.
fix: Bushes should stop catching bullets.
fix: Medals should show up on your mob again.
add: Added the Pflager Katsumata Series-D Blaster(M2019 Blaster), the LAPD .44 special (M2049 Blaster), and the PKL 'Double' Blaster from Bladerunner.
add: Added the RXF-M5 EVA pistol from Alien, as it appears on the Nostromo.
fix: You can now cut open boxing gloves with any sharp item.
imageadd: Made all the landing signs 2.5d kinda. Also fixed some inconsistencies with the rack and bed sprites.
imageadd: Resprited or added: rock sprites and rock prop code, a new set of sandy auto-turfs, generic curtains, generic pressure tank props, redone cargo tug sprites, beer coolers, the new varadero landing sign, entirely redone r&d props (lathe, analyzer, circuit printer), levers for conveyors, mulebots, red folding chairs and red stools, o2 storage racks, pepper spray refill thing, nuke pinpointer, deployable security barrier sprites, handheld mining drill sprites, signal wands, the EFTPOS scanner, plastic sheets, cardboard sheets, the double barrel shotgun, the M43 hummingbird pistol, motion device 3, the structural analyzer, the old flashbulb, EMP grenades, the voice component doodad, autopsy scanner, toner cartridge, paint buckets, robot analyzer, shield components, destination taggers, wrapping paper, game set, bike horn, mateba key, and another set of pews.

/:cl:

### CODE REVIEW

**Comment by Agameofscones** [comment]
gotta clean up some odd ends.

**Comment by MusselMain** [comment]
Damn, I knew energy/laser tech existed in-lore, but I didn't know we actually had any in-game (aside from the CAS mounted kind).

**Comment by Agameofscones** [comment]
forgot to fix db code.

File: code/game/machinery/recharger.dm [hunk 1]
```diff
@@ -10,7 +10,7 @@ obj/structure/machinery/recharger
 	active_power_usage = 15000	//15 kW
 	var/obj/item/charging = null
 	var/percent_charge_complete = 0
-	var/list/allowed_devices = list(/obj/item/weapon/melee/baton, /obj/item/cell, /obj/item/weapon/gun/energy/taser, /obj/item/device/defibrillator)
+	var/list/allowed_devices = list(/obj/item/weapon/melee/baton, /obj/item/cell, /obj/item/weapon/gun/energy, /obj/item/device/defibrillator)
```
Comment (fira) [inline code/game/machinery/recharger.dm:13]: There's still checks further down in process() of recharger to taser's. So my understanding is that this makes you able to put other energy guns in (including pred rifle/casters), but they don't actually charge

File: code/game/objects/items/props/rocks.dm [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+/obj/structure/prop/rock
+	name = "rock(s)"
+	icon = 'icons/obj/structures/props/rocks.dmi'
+	icon_state = "rock"//go figure
+	desc = "A solidified collection of local minerals. When melted, becomes a substance best known as lava."
+
+	opacity = 0
+	density = 1
+	var/dir_list_full = list(1,2,4,8,5,6,9,10)
+	var/dir_list_limited = list(1,2,4,8)
```
Comment (fira) [inline code/game/objects/items/props/rocks.dm:16]: This really should be the actual direction defines (NORTH, ...)

File: code/game/turfs/auto_turf.dm [hunk 3]
```diff
@@ -103,6 +103,22 @@
 	variant = 1
 	variant_prefix_name = "rocky"
 
+/turf/open/auto_turf/sand_white
+	layer_name = list("aged igneous", "wind swept dunes", "warn a coder", "warn a coder", "warn a coder")
+	icon_state = "varadero_1"
+	icon_prefix = "varadero"
+
+turf/open/auto_turf/sand_white/get_dirt_type()
```
Comment (fira) [inline code/game/turfs/auto_turf.dm:111]:
```suggestion
/turf/open/auto_turf/sand_white/get_dirt_type()
```

File: code/modules/projectiles/guns/revolvers.dm [hunk 4]
```diff
@@ -344,6 +344,79 @@
...
+// Blade Runner Blasters.
+/obj/item/weapon/gun/revolver/m44/custom/pkd_special
+	name = "\improper M2019 Blaster"
+	desc = "Properly known as the Pflager Katsumata Series-D Blaster, the M2019 is a relic of a handgun used by detectives and blade runners, having replaced the snub nose .38 detective special in 2019. Fires .44 custom packed sabot magnum rounds. Legally a revolver, the unconventional but robust internal design has made this model incredibly popular amongst collectors and enthusiasts."
+	current_mag = /obj/item/ammo_magazine/internal/revolver/m44/pkd
+	icon_state = "lapd_2019"
+	item_state = "m44"
+
+	fire_sound = "gun_pkd"
+	fire_rattle	= "sound/weapons/gun_pkd_fire01_rattle.ogg"
```
Comment (fira) [inline code/modules/projectiles/guns/revolvers.dm:357]:
```suggestion
	fire_rattle	= 'sound/weapons/gun_pkd_fire01_rattle.ogg'
```

**Review by fira — COMMENTED** [review]
Looks good to me overall, though i need to note that you might want to test specifically interactions with the pred rifle and caster which are also energy types, if not already

File: code/game/machinery/recharger.dm (hunk 1 above)
Comment (fira) [reply code/game/machinery/recharger.dm:13]: https://github.com/cmss13-devs/cmss13/blob/d51c243cdb55cda3068f318244234196452be797/code/game/machinery/recharger.dm#L76

**Comment by fira** [comment]
Any update? Aside from the quick fixes in suggests, all it needs is the charger being consistent (or kept as is, if that's what you intended)

**Comment by thwompa** [comment]
approval pending changes requested via discord

**Review by fira — APPROVED** [review]

**Comment by stanalbatross** [comment]
my brother in christ you have one icon conflict

**Review by Grumpuloso — APPROVED** [review]

**Pull request merged**